File: Solution/Python/771. Jewels and Stones.py
class Solution:
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        res = 0
        s = set([(i) for i in J])        
        for i in S:
            if i in s:
                res+=1
        return res
        

J = "aaAA"
S = "aAAbbbb"
s = Solution()
res = s.numJewelsInStones(J,S)
print(res)

File: Solution/Cpp/771._Jewels_and_Stones.cpp
#include <stdio.h>
#include <iostream>
#include <set>

using namespace std;

class Solution
{
  public:
    int numJewelsInStones(string J, string S)
    {
        int res = 0;
        set<char> s = set<char>();
        for (int i = 0; i < J.size(); i++)
        {
            s.insert(J[i]);
        }
        for (int i = 0; i < S.size(); i++)
        {
            if (s.find(S[i]) != s.end())
            {
                res++;
            }
        }
        return res;
    }
};

int main()
{
    Solution solution = Solution();
    string j = "hello";
    string s = "world";
    int res = solution.numJewelsInStones(j, s);
    std::cout << res << std::endl;
}